### PR TITLE
Add ReadMe

### PR DESCRIPTION
**Pull request by AndreiEvstratenko**
# Pull Request

## Description

Add ReadMe

Closes #1801

### CODE REVIEW

**Comment by brentchow** [comment]
This is a good start. If you don't mind, I'm going to clean this up.

**Comment by AndreiEvstratenko** [comment]
> This is a good start. If you don't mind, I'm going to clean this up.

Cool, of course!

**Comment by ox** [comment]
@AndreiEvstratenko would it be possible to add example usage code for each of the methods? I would like it to be super easy for developers to see how they can query the API and use the results quickly.

**Comment by brentchow** [comment]
Here's a good example of using code examples for methods in a readme https://github.com/Giphy/giphy-ios-sdk-core

File: README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+### Public functions
+
+#### *search*
+
+input parameters:
+
+- query: *String*
+- type: *[MediaType]?*
+- category: *String?*
```
Comment (brentchow) [inline README.md]: doesn't search also support `stereoscopicType`?

Comment (brentchow) [reply README.md]: it does https://developers.svrf.com/docs#operation/search

File: README.md [hunk 2]
```diff
@@ -1,105 +1,201 @@
...
 
-### Public functions
+### Search Endpoint
 
-#### *search*
+[The SVRF Search Endpoint][Docs Search] brings the power of immersive search found on [SVRF.com][SVRF] to your app or project. Our search engine enables your users to instantly find the immersive experience they're seeking. Content is sorted by the SVRF rating system, ensuring that the highest quality content and most relevant search results are returned first.
 
-input parameters:
+```swift
+// Add example
```
Comment (brentchow) [inline README.md]: @AndreiEvstratenko, can you add examples?

File: README.md [hunk 3]
```diff
@@ -1,105 +1,201 @@
...
 ## How to use
 
 ### Authentication
 
-1) Put your API key into .plist file for *"SVRF_API_KEY"* key. You can get API key on [Svrf](https://www.svrf.com).
-2) Put the following code into `didFinishLaunchingWithOptions` function in *AppDelegate*:
+1) Add your API key to the `.plist` file for *"SVRF_API_KEY"* key.
+2) Add the following code into `didFinishLaunchingWithOptions` function in *AppDelegate*:
 
 ```swift
```
Comment (brentchow) [inline README.md]: Can we expand this example to be more in-depth?

File: README.md [hunk 4]
```diff
@@ -1,105 +1,201 @@
...
+| stereoscopicType              | *String?*                                       |
 
-#### *getMedia*
+return *[Media]?*
 
-input parameters:
+### getMedia
 
-- id: *String*
+TODO: Add Description
```
Comment (brentchow) [inline README.md]: We need to add descriptions.

File: README.md (hunk 1 above)
Comment (AndreiEvstratenko) [reply README.md]: Search supports 'stereoscopicType'. I just missed it into readme.

File: README.md (hunk 2 above)
Comment (AndreiEvstratenko) [reply README.md]: Yeah, I added examples, check it please. I want to be sure that enough. I've added how to use this functions without extra code.

File: SvrfSDK/Source/SvrfSDK.swift [hunk 5]
```diff
@@ -175,7 +175,7 @@ public class SvrfSDK: NSObject {
         return nil
     }
     
-    public static func setBlendshapes(blendShapes: [ARFaceAnchor.BlendShapeLocation : NSNumber], for node: SCNNode) {
+    public static func setBlendShapes(blendShapes: [ARFaceAnchor.BlendShapeLocation : NSNumber], for node: SCNNode) {
```
Comment (brentchow) [inline SvrfSDK/Source/SvrfSDK.swift]: @AndreiEvstratenko I changed this from `setBlendshapes` -> `setBlendShapes`

File: README.md [hunk 6]
```diff
@@ -163,74 +174,87 @@ SvrfSDK.getMedia(id: "547963", onSuccess: { media in
...
 | blendShapes                   | *[ARFaceAnchor.BlendShapeLocation : NSNumber]*  |
 | node                          | *SCNNode*                                       |
 
-return *SCNNode*
+#### Example
+
+Map blend shapes to a *SCNNode*'s morpher.
+
+```swift
+class FaceFilter: SCNNode, VirtualFaceContent {
```
Comment (brentchow) [inline README.md]: @AndreiEvstratenko How does this example look?

Comment (AndreiEvstratenko) [reply README.md]: looks great. I think that's very understandably

File: SvrfSDK/Source/SvrfSDK.swift (hunk 5 above)
Comment (AndreiEvstratenko) [reply SvrfSDK/Source/SvrfSDK.swift]: I see. Now I will write "Shapes" with a capital letter

File: SvrfSDK/Source/SvrfSDK.swift [hunk 7]
```diff
@@ -186,6 +250,14 @@ public class SvrfSDK: NSObject {
...
     
+    /**
+     The SVRF API allows you to access all of SVRF's ARKit compatible face filters and stream them directly to your app. Use the `getFaceFilter` method to stream a face filter to your app and convert it into a *SCNNode* in runtime.
+     
+     - parameters:
+       - with: The device to fetch the face geometry from.
+       - media: The *Media* to generate the face filter from. The *type* must be `_3d`.
+     - returns: SCNNode
+     */
     public static func getFaceFilter(with device: MTLDevice, media: Media) -> SCNNode {
```
Comment (brentchow) [inline SvrfSDK/Source/SvrfSDK.swift]: I don't think we need the `MTLDevice` parameter. I think we're applying the occluder to the wrong node. It should be applied to the `occluderNode`, not the `ARSCNFaceGeometry`

File: SvrfSDK/Source/SvrfSDK.swift [hunk 8]
```diff
@@ -220,7 +292,14 @@ public class SvrfSDK: NSObject {
...
-    //MARK: private functions
+    // MARK: private functions
+    /**
+     Renders a *SCNScene* from a *Media*'s glb file using *SvrfGLTFSceneKit*.
+
+     - parameters:
+        - media: The *Media* to return a *SCNScene* from.
+     - returns: SCNScene?
+     */
     private static func getSceneFromMedia(media: Media) -> SCNScene? {
```
Comment (brentchow) [inline SvrfSDK/Source/SvrfSDK.swift]: This should also return an error if there's a problem generating the `SCNNode`

File: SvrfSDK/Source/SvrfSDK.swift (hunk 7 above)
Comment (AndreiEvstratenko) [reply SvrfSDK/Source/SvrfSDK.swift]: Yeah, Brent, you are right. Yesterday I was researching this and currently I know what we need and what we don't need. I'll change that.

Comment (AndreiEvstratenko) [reply SvrfSDK/Source/SvrfSDK.swift]: Yes, you are right. I didn't understand all about 3d models structure, but I was researching about that a lot and now I fully understand about that and I've made occluder correctly.

File: SvrfSDK/Source/SvrfSDK.swift (hunk 8 above)
Comment (AndreiEvstratenko) [reply SvrfSDK/Source/SvrfSDK.swift]: Yea, I'll update readme after updating the project

File: SvrfSDK/Source/SvrfSDK.swift (hunk 7 above)
[the remark this answers is not in the document]
Comment (AndreiEvstratenko) [reply SvrfSDK/Source/SvrfSDK.swift]: Yea, I'll update this after updating the project

**Comment by ox** [comment]
@AndreiEvstratenko @brentchow btw, I don't see a definition for what the `SvrfError` struct contains in the readme. As a developer I'd like to know what details I'm given in an error object when something fails.

**Pull request merged**